src/Page/Form.purs

```diff
@@ -47,18 +54,105 @@ component =
   initialState :: State
   initialState =
     { form: Left "Loading..."
+    , values: mempty
     }
 
   eval :: Query ~> H.ComponentDSL State Query Message m
   eval (Initialize a) = a <$ do
+    let values = keysPage testPage
+    H.modify_ _ { values = values }
     void $ bitraverse_
       (\e -> H.modify _ { form = Left e })
       (\f -> eval $ EvalForm f a)
       (Right testPage)
 
-  eval (EvalForm form a) = a <$ do
-    H.modify_ _ { form = pure form }
-    pure unit
+  eval (EvalForm expr a) = a <$ do
+    let evaled' = evalPage get expr
+        get key = Data.Map.lookup key values
+        values = keysPage expr
+    H.modify_ _ { values = values }
+    case evaled' of
+      Left _ -> pure unit
```

Comment:
❓ This ignores any errors that we had when we're evaluating the form. What should we do in this case? Do we want to throw the message up in a toast or something? Do we log errors somewhere we can later take a look at?

Reply:
At this point an error is fatal, so I think we could replace the whole `case` with
```purescript
H.modify_ _ { form = { expr, evaled: _ } <$> evaled }
```
And update the type of our state so the `Left` is an actual error type